sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathClient.java

```diff
@@ -1182,6 +1210,56 @@ public Response<PathProperties> getPropertiesWithResponse(DataLakeRequestConditi
         }, LOGGER);
     }
 
+    /**
+     * Returns the resource's metadata and properties.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <!-- src_embed com.azure.storage.file.datalake.DataLakePathClient.getPropertiesWithResponse#PathGetPropertiesOptions-Duration-Context -->
+     * <pre>
+     * PathGetPropertiesOptions options = new PathGetPropertiesOptions&#40;&#41;.setUpn&#40;true&#41;;
+     *
+     * Response&lt;PathProperties&gt; response2 = client.getPropertiesWithResponse&#40;options, timeout,
+     *     new Context&#40;key2, value2&#41;&#41;;
+     *
+     * System.out.printf&#40;&quot;Creation Time: %s, Size: %d%n&quot;, response2.getValue&#40;&#41;.getCreationTime&#40;&#41;,
+     *     response2.getValue&#40;&#41;.getFileSize&#40;&#41;&#41;;
+     * </pre>
+     * <!-- end com.azure.storage.file.datalake.DataLakePathClient.getPropertiesWithResponse#PathGetPropertiesOptions-Duration-Context -->
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/get-blob-properties">Azure Docs</a></p>
+     *
+     * @param options {@link PathGetPropertiesOptions}
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return A response containing the resource properties and metadata.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<PathProperties> getPropertiesWithResponse(PathGetPropertiesOptions options, Duration timeout,
+                                                              Context context) {
+        Context newContext;
```

Comment:
Since this is being used in three different locations, I wonder if we could create a shared method.

```java
static Context addUpnHeader(Supplier<Boolean> upnHeaderValue, Context context) {
    Boolean value = upnHeaderValue.get();
    if (value == null) {
        return context;
    }

    HttpHeaders headers = new HttpHeaders();
    headers.set(X_MS_UPN, Boolean.toString(value));
    if (context == null) {
        return new Context(AddHeadersFromContextPolicy.AZURE_REQUEST_HTTP_HEADERS_KEY, headers);
    } else {
        return context.addData(AddHeadersFromContextPolicy.AZURE_REQUEST_HTTP_HEADERS_KEY, headers);
    }
}
```

And a call to it could be

```java
context = addUpnHeader(() -> (options == null) ? null : options.isUpn(), context);
```

Reply:
I like this a lot, thanks for the suggestion!